### PR TITLE
Document how to deploy RHTAP and how to run Jenkins jobs

### PR DESCRIPTION
**Pull request by chmeliik**

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export quay_install_enabled=false
+# In quay.io, go to Account Settings -> Generate Encrypted Password -> Docker Configuration
+# Paste the auth file content here
+export QUAY__DOCKERCONFIGJSON=
+# 1. Create a quay.io organization if you don't have one
+# 2. Go to the organization's page -> Applications and create an application
+# 3. Go to the application's page -> Generate Token
+#    Select these permissions: View all visible repositories
+#                              TODO ^ probably wrong
+export QUAY__API_TOKEN=
```
Comment (chmeliik) [inline .env.template:43]: Does anyone know what permissions the token needs? I haven't even figured what the token is for to be honest :smile:

File: .env.template [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# RHTAP devs may be able to provide credentials for a shared ACS instance
+export acs_install_enabled=false
+export ACS__CENTRAL_ENDPOINT=
+export ACS__API_TOKEN=
+
+#### Gitlab
+export gitlab_enabled=false
+# https://pipelinesascode.com/docs/install/gitlab/#create-gitlab-personal-access-token
+# TODO ^ is this right?
+export GITLAB__TOKEN=
```
Comment (chmeliik) [inline .env.template:56]: Does anyone know what permissions the token needs?

File: README.md [hunk 3]
```diff
@@ -22,6 +22,39 @@ In development mode, the pipeline scripts can be tested using local shell script
...
+   (see [.env.template](.env.template)) to point to your fork
+3. Go to the Developer Hub url in your RHTAP instance, sign in and create a new component
+    - Select the CI type that you want to test (Tekton, Jenkins, GitHub actions, ...)
+4. Trigger the pipeline that you want to test (typically by creating or merging a PR)
+    - Jenkins pipelines need some manual setup first, see [Configuring Jenkins](#configuring-jenkins)
+      and [Setting up Jenkins jobs](#setting-up-jenkins-jobs)
+
+### Deploying RHTAP for development
+
+You'll need an OpenShift cluster (version ... <!-- TODO supported versions? -->) for deploying RHTAP.
```
Comment (chmeliik) [inline README.md:40]: Where do we document what OpenShift versions RHTAP supports?

**Review by chmeliik — COMMENTED** [review]
Needs https://github.com/redhat-appstudio/rhtap-cli/pull/300 to work properly

File: .env.template (hunk 1 above)
Comment (chmeliik) [reply .env.template]: I remembered RHTAP has customer docs. (Also, wow I actually got the permissions right)

File: .env.template (hunk 2 above)
Comment (chmeliik) [reply .env.template]: Found the relevant [doc](https://docs.redhat.com/en/documentation/red_hat_trusted_application_pipeline/1.2/html-single/installing_red_hat_trusted_application_pipeline/index#optional_integrating_gitlab). Which still doesn't say what permissions the token needs. But hey, if it's good enough for RHTAP customers it's good enough for me ¯\\\_(ツ)\_/¯

File: README.md (hunk 3 above)
Comment (chmeliik) [reply README.md]: Found this as well

**Comment by chmeliik** [comment]
I think I've addressed the standout TODOs, though https://github.com/redhat-appstudio/rhtap-cli/pull/300 would still be needed in order for the instructions to work well

**Review by jduimovich — APPROVED** [review]
This is a good start. I do the same thing as .env except outside this repo so my dev shell has the correct vars installed. 
We should also add a default install rhtap config script which uses the same vars. I can do a PR for the one I use and we can evolve it.

**Comment by chmeliik** [comment]
> This is a good start. I do the same thing as .env except outside this repo so my dev shell has the correct vars installed. We should also add a default install rhtap config script which uses the same vars. I can do a PR for the one I use and we can evolve it.

The .env file is intended for use with https://github.com/redhat-appstudio/rhtap-cli/blob/main/integration-tests/scripts/install.sh (if that's the sort of install rhtap config script you had in mind)

File: .env.template (hunk 2 above)
Comment (xinredhat) [reply .env.template]: @chmeliik you can set all

**Pull request merged**